### ISSUE TITLE
Staging Review finding: Keyboard focus behaves unexpectedly

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#104799]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Ask VA Integration
**Product:** Ask VA
**Feature:** Ask Va

## Findings details
**VA.gov Experience Standard - issue:** [User experiences predictable interactions with components and patterns.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

There are a few places in the product where keyboard focus behaves unexpectedly after an interaction:
- On [/ask-va/introduction](https://staging.va.gov/contact-us/ask-va/introduction) when searching for a reference ID, focus stays on the search button (should go to the "Showing..." message or to the "We didn't find..." message)
- On [/ask-va/introduction](https://staging.va.gov/contact-us/ask-va/introduction) when authenticated and paginating through previously-asked questions, focus is lost (should go to the "Showing X - Y" message for each page)
- On [/ask-va/category-topic-1](https://staging.va.gov/contact-us/ask-va/category-topic-1) when unauthenticated and closing the login modal ("Back to category" button, X button, or [esc] key)

Focus management is important for a couple of reasons:
- For sighted keyboard-only users, what has keyboard focus determines where you're starting from as you tab through the page. If you close a modal and focus is lost, then the next time you press [tab] your viewport may jump unexpectedly and you may have to [tab] or [shift] + [tab] several times to get back to where you thought you were visually.
- For screen reader users, the same thing applies but with a little extra value --- when something receives focus programmatically, that element is immediately announced to the user. This is important when performing an action, as managing focus can ensure that screen reader users hear something to verify that an action took place or to provide information on the status of the action. Often when focus behaves unexpectedly, screen reader users don't hear anything at all. Silence can be disorienting and confusing, and users have to guess about what they're supposed to do next.

## Recommended action

When performing an action, keyboard focus should always go somewhere actionable:
- summary message for results displayed on screen
- alert indicating that an action is completed or if there's been an error
- back to the place you started, eg. when closing a modal focus returns to the button that opened the modal

The right place to go is context dependent, but the general rule is that you want to send your user to the place where they're most likely to want to be next.

Note that when managing focus programmatically, you can send focus to elements that wouldn't normally be focusable using `tabindex="-1"`.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Keyboard
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by arielma-tw** [comment]
Issues has been addressed in https://github.com/department-of-veterans-affairs/ask-va/issues/1660
